plugins/modules/ec2_launch_template.py

```diff
@@ -516,6 +532,13 @@ def create_or_update(module, template_options):
     out = {}
     lt_data = params_to_launch_data(module, dict((k, v) for k, v in module.params.items() if k in template_options))
     lt_data = scrub_none_parameters(lt_data, descend_into_lists=True)
+
+    if lt_data.get('MetadataOptions'):
+        if not module.boto3_at_least('1.20.30'):
+            lt_data['MetadataOptions'].pop('InstanceMetadataTags')
+        if not module.boto3_at_least('1.18.29'):
+            lt_data['MetadataOptions'].pop('HttpProtocolIpv6')
```

Comment:
So in other modules it seems to be more standard to fail or warn that the required boto version isn't used when passing a particular param. Personally I find an explicit fail or warn better to work with so that it's clear on the what's required to resolve etc

Reply:
I've added and pushed minutes ago :)

>  I find an explicit fail 

I'm not sure if we should fail, because the boto3 requirements for the entire collection is lower.  
Furthermore, when we fail, all playbooks that run on 3.0.1, won't run with 3.1.0 anymore when the boto3 library is not updated - just because of parameters that most likely are not requested.

Reply:
Maybe we can fail, when other values as the APIs default values are requested ...

Reply:
some more opinions cc @alinabuzachis  @tremble  @jillr

Reply:
Awesome think I hadn't refreshed 😀  

Yeah that makes sense it is a little inconsistent across the modules to be fair as some places do seem to explicitly fail and others warn but yeah definitely better to have something to flag it up

Reply:
I can suggest something like this: if ignoring the new options won't affect the normal behaviour of the module, I'd probably only warn.

Reply:
Not entirely sure of how handle that and I agree with what @marknet15 said.

However, there is an example https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/ec2_instance.py#L989

Reply:
Ah, there is a `reason` parameter. Thx

Reply:
>  if ignoring the new options won't affect the normal behaviour of the module, I'd probably only warn.

Now it 

* failed if `http_protocol_ipv6: enabled` and `instance_metadata_tags: enabled` are requested but the boto3 version is not sufficient
* passed if  `http_protocol_ipv6: disabled` and `instance_metadata_tags: disabled` are requested or not set (default values) and boto3 version is not sufficient. That covers the backwart compatiblity from community.aws < 3.1.0
* passed if  `http_protocol_ipv6: enabled` and `instance_metadata_tags: enabled` are requested  and boto3 version is sufficient.

All three cases are covered in the integration test